packages/node/src/lib/novu.ts

```diff
@@ -40,8 +41,25 @@ export class Novu extends EventEmitter {
   readonly organizations: Organizations;
   readonly workflowOverrides: WorkflowOverrides;
 
-  constructor(apiKey: string, config?: INovuConfiguration) {
+  constructor(config?: INovuConfiguration);
+  constructor(apiKey: string, config?: INovuConfiguration);
+  constructor(...args: any) {
     super();
+
+    let apiKey: string | undefined;
+    let config: INovuConfiguration | undefined;
+
+    if (arguments.length === 2) {
+      apiKey = args[0];
+      config = args[1];
+    } else if (arguments.length === 1) {
+      const { apiKey: key, ...rest } = args[0];
+      apiKey = key;
+      config = rest;
+    } else {
+      apiKey = getEnvVariable('NOVU_API_KEY');
+    }
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
**note (non-blocking):** Should we add a unit test for this behaviour? Just so that we are confident in this tricky part here‏

Reply:
Done.